pkg/workflow/providers/oam/apply.go

```diff
@@ -146,10 +206,82 @@ func lookUpValues(v *value.Value) (*common.ApplicationComponent, *value.Value, s
 	return comp, patcher, clusterName, overrideNamespace, env, nil
 }
 
+func (p *provider) loadDynamicComponent(comp *common.ApplicationComponent) (*common.ApplicationComponent, error) {
+	if comp.Type != "ref-objects" {
+		return comp, nil
+	}
+	_comp := comp.DeepCopy()
+	props := &struct {
+		Objects []struct {
+			APIVersion string            `json:"apiVersion"`
+			Kind       string            `json:"kind"`
+			Name       string            `json:"name,omitempty"`
+			Selector   map[string]string `json:"selector,omitempty"`
+		} `json:"objects"`
+	}{}
+	if err := json.Unmarshal(_comp.Properties.Raw, props); err != nil {
+		return nil, errors.Wrapf(err, "invalid properties for ref-objects")
+	}
+	var objects []*unstructured.Unstructured
+	addObj := func(un *unstructured.Unstructured) {
+		un.SetResourceVersion("")
+		un.SetGeneration(0)
+		un.SetOwnerReferences(nil)
+		un.SetDeletionTimestamp(nil)
+		un.SetManagedFields(nil)
+		un.SetUID("")
+		unstructured.RemoveNestedField(un.Object, "metadata", "creationTimestamp")
+		unstructured.RemoveNestedField(un.Object, "status")
+		// TODO(somefive): make the following logic more generalizable
+		if un.GetKind() == "Service" && un.GetAPIVersion() == "v1" {
```

Comment:
NB: in addition to "Service", any built-in or custom kubernetes types w/ defaulting or immutability will need the hacks below, that's why i was proposing using work api to deploy resource (in another place)